### DISCUSSION TITLE
Time-dependent diffusivity

### DISCUSSION THREAD

**Discussion by nreuge** [zwicker-group/py-pde#188] [Q&A]
Dear Dr. Zwicker,

I'm considering about using your package for my work involving modeling the diffusion of chloride ions into concretes.
I've a important question: is it possible using py-pde with a time-dependent diffusivity D=D(t), the time-dependence law being known? (in concretes, the diffusity progressively decreases over the years...)
Or would I have to chain several simulations, the one after the other with D constant but different, to take into account this time-dependency? I'd like to avoid this method, it wouldn't be really convenient...

Congratulations for your work!

Any help would be appreciated, than you!

**Comment by david-zwicker** [comment]
Yes, a time-dependent diffusivity should not pose any problems. It's not currently implemented in the [`DiffusionPDE` class](https://py-pde.readthedocs.io/en/latest/packages/pde.pdes.diffusion.html?highlight=DiffusionPDE#pde.pdes.diffusion.DiffusionPDE), but you can simply write a diffusion equation with a time-dependent diffusivity using the [`PDE` class](https://py-pde.readthedocs.io/en/latest/packages/pde.pdes.pde.html#pde.pdes.pde.PDE).
